packages/contracts-bedrock/src/Safe/VetoerSet/OwnerGuard.sol

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { FixedPointMathLib } from "solady/utils/FixedPointMathLib.sol";
+
+import { Safe } from "safe-contracts/Safe.sol";
+import { BaseGuard } from "safe-contracts/base/GuardManager.sol";
+import { Enum } from "safe-contracts/common/Enum.sol";
+
+import { ISemver } from "src/universal/ISemver.sol";
+
+/// @title OwnerGuard
+/// @notice This Guard contract is used to enforce a maximum number of owners and a required
+///         threshold for the Safe Account.
+contract OwnerGuard is ISemver, BaseGuard {
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0
+    string public constant version = "1.0.0";
+
+    /// @notice The Safe Account for which this contract will be the guard.
+    Safe public immutable safe;
+
+    /// @notice The maximum number of owners. Can be changed by supermajority.
+    uint8 public maxOwnerCount;
+
+    /// @notice Thrown at deployment if the current Safe Account owner count can't fit in a `uint8`.
+    /// @param ownerCount The current owner count.
+    error OwnerCountTooHigh(uint256 ownerCount);
+
+    /// @notice Thrown if the new owner count is above the `maxOwnerCount` limit.
+    /// @param ownerCount The Safe Account owner count.
+    /// @param maxOwerCount The current `maxOwnerCount`.
+    error InvalidOwnerCount(uint256 ownerCount, uint256 maxOwerCount);
```

Comment:
I would actually prefer to swap these two errors, since I prefer `Invalid` for defying a machine construct and `TooHigh` for it being above the `maxOwnerCount`